src/Microsoft.AspNet.Authentication/AuthenticationHandler.cs

```diff
@@ -62,8 +63,8 @@ protected string CurrentUri
         /// <param name="options">The original options passed by the application control behavior</param>
         /// <param name="context">The utility object to observe the current request and response</param>
         /// <param name="logger">The logging factory used to create loggers</param>
-        /// <returns>async completion</returns>
-        public async Task InitializeAsync(TOptions options, HttpContext context, ILogger logger, IUrlEncoder encoder)
+        /// <returns>True if initialization was successful</returns>
+        public async Task<bool> InitializeAsync(TOptions options, HttpContext context, ILogger logger, IUrlEncoder encoder)
```

Comment:
It doesn't look like InitializeAsync is ever overridden. The flow control comes from HandleRequestAsync insteadl. I think you can revert this.